www/js/lib/zimArchive.js

```diff
@@ -50,75 +50,62 @@ define(['zimfile', 'zimDirEntry', 'util', 'utf8'],
-                    var directoryPath = regexpDirectoryPathResults[1];
-                    // We need to find all the actual files that belong to the same archive :
-                    // they are in the same directory and have the same prefix
-                    // TODO : this seems to scan the whole DeviceStorage instead of only the directory?
-                    var cursor = storage.enumerate(directoryPath);
-                    cursor.onsuccess = function () {
-                        if (this.result) {
-                            var file = this.result;
-                            // TODO if it's not possible to avoid scanning sub-directories,
-                            // we should take care we don't match files inside those sub-directories
-                            if (file && file.name && file.name.startsWith(fileNamePrefix)) {
-                                fileArray.push(file);
-                            }                            
-                        }
-                        if (this.done) {
-                            return zimfile.fromFileArray(fileArray).then(function (zimFile) {
-                                that._file = zimFile;
-                                callbackReady(that);
-                            });
-                        }
-                        else {
-                            this.continue();
-                        }
-                    };
-                    cursor.onerror = function (error) {
-                        alert("Impossible to find the actual files corresponding to " + path + " : " + error);
-                    };
-                }
-                else {
-                    alert("Impossible to find the directory of " + path);
-                }
-            }, function(error) {
-                alert("Error reading ZIM file " + path + " : " + error);
-            });
+            var p;
+            if (/.*zim..$/.test(path)) {
+                // splitted archive
+                that._searchArchiveParts(storage, path.slice(0, -2)).then(function(fileArray) {
+                    createZimfile(fileArray);
+                }, function(error) {
+                    alert("Error reading files in splitted archive " + path + ": " + error);
+                });
+            }
+            else {
+                storage.get(path).then(function(file) {
+                    createZimfile([file]);
+                }, function(error) {
+                    alert("Error reading ZIM file " + path + " : " + error);
+                });
+            }
         }
     };
 
     /**
+     * Searches the directory for all parts of a splitted archive.
+     * @param {Storage} storage storage interface
+     * @param {String} prefixPath path to the splitted files, missing the "aa" / "ab" / ... suffix.
+     * @returns {Promise} that resolves to the array of file objects found.
+     */
```

Comment:
Cool : this is a much cleaner and concise way to do it.
It appears to work well on my device.
I'll just remove the unused variable and debug log